### DISCUSSION TITLE
Can't get 'git lfs' to work with --shared repos on local server

### DISCUSSION THREAD

**Discussion by imrep** [git-lfs/git-lfs#5791] [Q&A]
Hi,
I'm stuck. Any help is appreciated!
Configuration:
Debian 18.04LTS
Git 2.17.1
git-lfs/3.2.0

1. on server
$git init Test.git --bare
in Test.git folder
$git lfs install

on client (windows11 with git version 2.45.2.windows.1, git-lfs/3.5.1 (GitHub; windows amd64; go 1.21.7; git e237bb3a))
from git bash
$git init ClientTest
$cd ClientTest
$git remote add origin xxx@xxxx:/var/git/Test.git
$git lfs track *.bin
add test.bin file into working folder
$git add test.bin
$git commit -m "test"
$git push origin master
$ git push origin master
Locking support detected on remote "origin". Consider enabling it with:
  $ git config lfs.https://xxxx//var/git/Test.git/info/lfs.locksverify true
Uploading LFS objects: 100% (1/1), 3 B | 0 B/s, done.
Enumerating objects: 5, done.
Counting objects: 100% (5/5), done.
Delta compression using up to 36 threads
Compressing objects: 100% (2/2), done.
Writing objects: 100% (3/3), 345 bytes | 345.00 KiB/s, done.
Total 3 (delta 0), reused 0 (delta 0), pack-reused 0 (from 0)
To xxxx:/var/git/Test.git
   4c34fcb..98291c2  master -> master

All working fine
Now when i create server repository with "--shared" parameter and repeat all steps. I get.

Locking support detected on remote "origin". Consider enabling it with:
  $ git config lfs.https://xxxx//var/git/Test.git/info/lfs.locksverify true
got status 500 when uploading OID f703b8615ec2d12d68c2b8a13444ecbf662297cb0000e201f1d1f2f5f7ad3bbd: internal error: fatal: I/O error: Permission denied (os error 13) at path "/var/git/Test.git/lfs/incomplete/f703b8615ec2d12d68c2b8a13444ecbf662297cb0000e201f1d1f2f5f7ad3bbdlUodQdzOUEMS"
error: failed to push some refs to 'xxxx:/var/git/Test.git'

It does not work even when i run "chmod -R 777 *" in Test.git folder on server.

Best regards,
Imre

**Comment by chrisd8088** [comment]
Hey, I'm sorry you're having trouble.  I'm afraid that we may not be able to help too much as this project is for the Git LFS client specifically, and it sounds like you're having a problem with some kind of Git LFS server, of which there are quite a few.

You don't describe the server-side setup, but I'm guessing you're trying to call `git init` with the `--shared` parameter you mention?  You may want to review Git's [documentation](https://git-scm.com/docs/git-init#Documentation/git-init.txt-code--sharedcodecodecodecodefalsecodecodetruecodecodeumaskcodecodegroupcodecodeallcodecodeworldcodecodeeverybodycodeemltpermgtem) of that command option.  I'd also recommend getting in touch with the maintainers of the Git LFS server project you are using.

(If by chance that's the `lfs-test-server`, note that that's a test/demo server we maintain but don't guarantee will work in all circumstances, and which is not production-ready.  If that's what you are using, please provide a full description of your environment and operating system, and the complete set of commands you are using on the server side to create repositories and configure and run the server, and maybe we can help a bit.)

**Reply by imrep** [reply]
Thanks for the reply. I had no idea there were dedicated servers for Git LFS. I apologize for my lack of experience. I had an idea that one client connected over SSH to another machine, and used to run commands or upload files.
On remote machine i have git-lfs-transfer installed.

Remote machine:
git version 2.25.1
git-lfs/3.5.1 (GitHub; linux amd64; go 1.21.8)

Local machine:
git version 2.45.2.windows.1
git-lfs/3.5.1 (GitHub; windows amd64; go 1.21.7; git e237bb3a)

However, I encountered an issue: the permissions in the lfs/ subdirectory of my Git repository were preventing file creation. Specifically, the directories created in .git/lfs/objects/XX/YY lacked the necessary permissions. When I added execute permission to these directories and reran the command on the client machine, I made progress, until file was created successfully.
As i mentioned before this error happens only when repository is created with --shared flag.

**Reply by chrisd8088** [reply]
It sounds like you've got it working, so that's good!

The Implementations [page](https://github.com/git-lfs/git-lfs/wiki/Implementations) of our wiki lists some of the available Git LFS server options, if you want to explore any of those in the future to see if they suit your needs.
